platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -1067,13 +1078,22 @@ private void runWithSnapshotExecutor(
      * Write all hashes to hashStore
      */
     private void writeHashes(final long maxValidPath, final Stream<VirtualHashRecord> dirtyHashes) throws IOException {
-        if ((dirtyHashes == null) || (maxValidPath <= 0)) {
+        if (hasDiskStoreForHashes) {
+            if (maxValidPath < 0) {
+                // Empty store
+                hashStoreDisk.updateValidKeyRange(-1, -1);
+            } else {
+                hashStoreDisk.updateValidKeyRange(0, maxValidPath);
+            }
+        }
```

Comment:
This is another problem, although related to the bug. Before this change, if `maxValidPath` was -1 (that is, if the virtual map becomes empty), `hashStoreDisk` and the corresponding index were not updated at all. It led to a few memory chunks to remain unused in the index. Not a memory leak, since these chunks would be reused once the map grows, but still not nice